rust/blockstore/src/types.rs

```diff
@@ -356,3 +356,210 @@ impl<
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
```

Comment:
Would prefer if state machine test was moved out into its own file given the size